halo2/src/circuit_builder.rs

```diff
@@ -251,31 +339,24 @@ pub(crate) fn analyzed_to_circuit<T: FieldElement>(
         })
         .collect();
 
-    let witness_cols = cd
-        .witness
+    let witness_cols = witness
         .iter()
         .enumerate()
         .map(|(n, (name, _))| (name.to_string(), (ColumnKind::Fixed, n)));
 
     let wit = Witness {
-        num_rows: cd.witness.len(),
+        num_rows: witness.len(),
         columns: witness_cols
             .map(|(name, _)| ColumnWitness::new(name, 0))
             .collect(),
-        witness,
+        witness: converted_witness,
     };
 
     // build plaf. -------------------------------------------------------------------------
 
     let plaf = Plaf {
-        info,
-        columns,
-        polys,
-        metadata: Default::default(),
-        lookups,
-        shuffles,
         copys: copies,
-        fixed,
+        ..plaf
```

Comment:
I think this doesn't change anything, the `plaf` already has the copy constraints, no? That's good, then we can get rid of the code duplication :) We just only still need to compute the publics...

Reply:
will check